src/mocks/mock_vault.cairo

```diff
@@ -0,0 +1,96 @@
+use core::starknet::event;
```

Comment:
why did you create this contract? Vault is a component and should be tested directly not through a mock see how openzeppelin does it: https://github.com/OpenZeppelin/cairo-contracts/blob/5f03fba338b4cb321776947fb98b4584a32adece/src/tests/token/erc20/test_erc20.cairo#L24

Reply:
don't we anyway need a mock contract. It's just that we can also test without deploying it. Are you suggesting I should not deploy it and do it like [this](https://book.cairo-lang.org/ch16-02-03-testing-components.html#testing-components-without-deploying-a-contract)

Reply:
You can use contract_state_for_testing https://github.com/OpenZeppelin/cairo-contracts/blob/5f03fba338b4cb321776947fb98b4584a32adece/src/tests/token/erc20/test_erc20.cairo#L24

Reply:
It is better to use the component state for ERC20 and 721 too, right?

Reply:
Something like this:
```
fn VAULT_STATE() -> VaultComponentState {
    PwnVaultComponent::component_state_for_testing()
}

type ERC20ComponentState = ERC20Component::ComponentState<MockERC20::ContractState>;

fn ERC20_STATE() -> ERC20ComponentState {
    ERC20Component::component_state_for_testing()
}

type ERC721ComponentState = ERC721Component::ComponentState<MockERC721::ContractState>;

fn ERC721_STATE() -> ERC721ComponentState {
    ERC721Component::component_state_for_testing()
}
```

Reply:
Yes you can do this, or you can deploy the tokens from the mocks.

Reply:
Actually let me have a closer look to this test since it’s a peculiar with 3 separate components

Reply:
Ok I think you should follow starknet foundry docs on how to test without deploying but you should deploy ERC20 and erc721 from the mocks because there are used as lended asset and collateral